rust/agama-lib/src/localization/store.rs

```diff
@@ -1,51 +1,150 @@
 //! Implements the store for the localization settings.
 // TODO: for an overview see crate::store (?)
 
-use super::{LocalizationClient, LocalizationSettings};
+use super::{LocalizationHTTPClient, LocalizationSettings};
 use crate::error::ServiceError;
-use zbus::Connection;
+use crate::localization::model::LocaleConfig;
 
 /// Loads and stores the storage settings from/to the D-Bus service.
-pub struct LocalizationStore<'a> {
-    localization_client: LocalizationClient<'a>,
+pub struct LocalizationStore {
+    localization_client: LocalizationHTTPClient,
 }
 
-impl<'a> LocalizationStore<'a> {
-    pub async fn new(connection: Connection) -> Result<LocalizationStore<'a>, ServiceError> {
+impl LocalizationStore {
+    pub async fn new() -> Result<LocalizationStore, ServiceError> {
         Ok(Self {
-            localization_client: LocalizationClient::new(connection).await?,
+            localization_client: LocalizationHTTPClient::new().await?,
```

Comment:
When `LocalizationHTTPClient::new` is fixed, this method won't need to be async.